sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedProcessorContextImpl.java

```diff
@@ -23,4 +24,17 @@ public String getLeaseToken() {
 
         return changeFeedObserverContext.getLeaseToken();
     }
+
+    @Override
+    public CosmosDiagnosticsContext getDiagnostics() {
+        if (changeFeedObserverContext == null) {
+            throw new IllegalStateException("changeFeedObserverContext cannot be null!");
+        }
+
+        if (changeFeedObserverContext.getFeedResponse() == null) {
+            throw new IllegalStateException("feed response cannot be null!");
+        }
+
+        return changeFeedObserverContext.getFeedResponse().getCosmosDiagnostics().getDiagnosticsContext();
```

Comment:
getComsosDiagnostics() - can this ever be null?

Reply:
Yeah I have a null check locally, didn't push - tests failing anyways because of the wiring issue want to fix that first

Reply:
Added